app/models/post.rb

```diff
@@ -1,19 +1,43 @@
-class Post < ApplicationRecord
-  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
-  has_many :likes, class_name: 'Like', dependent: :destroy
-  has_many :comments, dependent: :destroy
+# spec/models/post_spec.rb
 
-  validates :title, presence: true, length: { maximum: 250 }
-  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
-  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+require 'rails_helper'
 
-  after_save :update_post_counter
+RSpec.describe Post, type: :model do
+  it 'is valid with valid attributes' do
+    user = User.create(username: 'testuser')
+    post = Post.new(title: 'Test Post', author: user)
+    expect(post).to be_valid
+  end
+
+  it 'is not valid without a title' do
+    post = Post.new(title: nil)
+    expect(post).to_not be_valid
+  end
+
+  it 'is not valid if title length exceeds 250 characters' do
+    post = Post.new(title: 'a' * 251)
+    expect(post).to_not be_valid
+  end
+
+  it 'should have a comments_counter that is a non-negative integer' do
+    post = Post.new(title: 'Test Post')
+    expect(post.comments_counter).to be >= 0
+    expect(post.comments_counter).to be_an(Integer)
```

Comment:
- [ ] Tests are still failing, getting the below error with the post model. While fixing the issues as requested by the previous reviewer you deleted the `Post` model class and added some test cases there. You should not do that it will break the entire application. Please make sure to revert the change in the `Post` model and all tests should pass without any errors.

![image](https://github.com/AlyAtalla/blog-app/assets/83803414/2a5953fe-8da5-44e4-9673-d9123b6c8d3c)

Reply:
thank you